### DISCUSSION TITLE
Read out oil tank meters... / Öltank Anzeigen auslesen...

### DISCUSSION THREAD

**Discussion by tbit-c** [jomjol/AI-on-the-edge-device#916] [Q&A]
Hey guys,
{english}
we have an oil heating system at home (happy not having gas these days...) and there are 6 tanks each with a separate gauge to read the volume. They all look the same like this:

![IMG_5464](https://user-images.githubusercontent.com/69671358/184949867-e2188059-e744-4df1-b2cf-d08bba39e769.jpeg)

Is there any chance to also read out these and would that be possible with the edge device from this project here?
Thanks for some comments.

{Deutsch}
wir haben eine Ölheizung bei uns zu Hause (zum Glück in diesen Tage) und diese wird gespeist von 6 Tanks. Jeder dieser Tanks hat eine Anzeige mit der die Menge abgelesen werden kann. Diese schauen so aus:

![IMG_5464](https://user-images.githubusercontent.com/69671358/184951184-15376ec8-75de-4270-a1ea-0bf4a8443b73.jpeg)

Gibt es eine Möglichkeit mit Hilfe diese Projektes die Anzeigen auszulesen?
Ich bekomme das irgendwie nicht hin.

Vielen Dank und Grüße

Tom

**Comment by jomjol** [comment]
Hi Tom,

das geht mit dem jetzigen Setup nicht. Du braucht ein völlig neues Set an Trainingsdaten und auch die Logik müsste angepasst werden. Aufwand ist ziemlich hoch.

Gruß,
'  jomjol

**Reply by tbit-c** [reply]
Hey Jomjol,
Damn!... Hab ich mir schon fast gedacht... Liegt verm. daran, dass die Zahlen so dämlich im Kreis angeordnet sind.
Hast Du ne Schätzung, über welches Stundenkontingent wir sprechen?
Vielen Dank auf jeden Fall für Deine schnelle Antwort.
Cheers, Tom

**Reply by jomjol** [reply]
Denke, da bin ich schnell im zweistelligen Bereich. Auch das Erzeugen der Trainingsdaten ist nicht ganz trivial, da du eigentlich von jedem Zählerstand grob ein Bild brauchst. Müsstest also warten bis alles mal leer war ... oder die Bilder per Bildverarbeitung selbst "zeichnen" ist ziemlich viel Aufwand.

**Reply by Koregan82** [reply]
Könnte die KI/Bilderkennung nicht einfach den Winkel des Zeigers bestimmen und als Winkel ausgeben?
Dann könnte man über eine Zuordnung Startwinkel/Endwinkel und werte nen Wert berechnen lassen, ohne die Zahlen der Anzeige auswerten zu müssen. 
Vielleicht denke ich auch falsch, wäre aber mein Gedanke hierzu.
Ich wäre auch sehr daran interessiert, da ich auch unterschiedliche Heizöltanks gerne mit dem ESP überwachen würde.

**Comment by jomjol** [comment]
Klar funktioniert jetzt ja auch nicht anders: Der Winkel des Zeigers wir in einen Wert zwischen 0 und 10 umgewandelt.

Aber die Logik für Min- / Max-Winkel müsste implementiert werden (kannst du aber auch offline in deinem Server berechnen).
Und viel wichtiger: für diese Art von Zeiger brauchst du ein eigents trainiertes Netz, es unterscheidet sich zu sehr von den anderen Zeigern. Daher brauchst du von jedem Zeigerstand und jeder Zeigerart ein Bild mit Label und ein entsprechendes Training.

**Comment by tbit-c** [comment]
Hi jomjol,

Ich bins mal wieder nach sehr langer Zeit… ich hatte leider überhaupt keine Zeit mich drum zu kümmern…
Gibts von Deiner Seite schon news? Ich habe im Web / Github nix gesehen in der Richtung.

Vorschlag von meiner Seite wäre, dass ich so eine Uhr bei Roth besorge und ich Deiner Vorgabe die ganzen Bilder erstelle.
Was denkst Du, kann ich denn finanziell mal einplanen, wenn ich Dich um die Umsetzung der Lösung bitten würde und ich Dir die Bilder bereit stelle?

Vielen Dank und Grüße
Tom
